basic_system/src/system_functions/modexp/mod.rs

```diff
@@ -221,12 +229,23 @@ fn modexp_as_system_function_inner<
     );
 
     #[cfg(not(any(all(target_arch = "riscv32", feature = "proving"), test)))]
-    let output = ::modexp::modexp(
-        base.as_slice(),
-        exponent.as_slice(),
-        modulus.as_slice(),
-        allocator,
-    );
+    let output = if USE_DELEGATION {
```

Comment:
Can we run delegation implementation depending on proving feature, so we can remove this extra argument?

And actually maybe it makes sense to rename delegation implementation to something like "proving implementation"/"advice_implementation"? As I understand it uses delegation depending on target

Reply:
I wanted to avoid using the feature, as we want to use the basic_system crate in both modes for the same binary. Otherwise, we'll need to somehow import the crate twice for server to run both the sequencing system and the prover input gen. 

The naming is just because it uses the modexp::delegation implementation, but can rename to advice, it makes more sense.